libkbfs/block_journal.go

```diff
@@ -1105,6 +1138,95 @@ func (j *blockJournal) ignoreBlocksAndMDRevMarkers(ctx context.Context,
 	return nil
 }
 
+func (j *blockJournal) saveBlocksUntilNextMDFlush() error {
+	// Copy the current journal entries into a new journal.  After the
+	// next MD flush, we can use the saved journal to delete the block
+	// data for all the entries in the saved journal.
+	first, err := j.j.readEarliestOrdinal()
+	if os.IsNotExist(err) {
+		return nil
+	} else if err != nil {
+		return err
+	}
+
+	last, err := j.j.readLatestOrdinal()
+	if os.IsNotExist(err) {
+		return nil
+	} else if err != nil {
+		return err
+	}
+
+	savedJournal := j.saveUntilMDFlush
+	if savedJournal == nil {
+		savedJournalDir, err := ioutil.TempDir(j.dir, "saved_block_journal")
```

Comment:
this has to persist across restarts, right? shouldn't it then be a predictable name, and initialized on startup? (if it exists) Or is that for a future PR?